### ISSUE TITLE
Levelling tracker and SkillTree cal

### ISSUE THREAD

**Issue by zefrogi** [Lailloken/Lailloken-UI#345]
It is updated to the latest version, and I am having trouble with the levelling tracker. It simply won't show. I've tried fresh installs of the UI and the AutoKeys. The tracker button changes colour, but the usual UI above the exp bar never shows.

The screen check for Skilltree also seems to be time-out super fast, so I'm never able to capture it. Not sure if this may be related to the first issue.

Side note: I do love the tool! Do you know if there's any way to make it run when launching PoE with Steam? 

Edit: Tested v1.52.0 and that shows the levelling UI. I have tried the latest version again and it manage to do the Skilltree capture the levelling window still doesn't show.

**Comment by Lailloken** [comment]
> Edit: Tested v1.52.0 and that shows the levelling UI. I have tried the latest version again and it manage to do the Skilltree capture the levelling window still doesn't show.

v1.52.1 made the leveling overlay movable, so maybe that's the issue. Are you a first-time user (i.e. was it a v1.52.1 clean install)?

> The screen check for Skilltree also seems to be time-out super fast

The Windows snipping tool shouldn't time out by itself, it only closes after user input.

**Comment by zefrogi** [comment]
> v1.52.1 made the leveling overlay movable, so maybe that's the issue. Are you a first-time user (i.e. was it a v1.52.1 clean install)?
Long-timer use, but I have done a fresh install of the v1.52.1 UI and auto keys.

> The Windows snipping tool shouldn't time out by itself, it only closes after user input.
The UI reports "Screen cap failed" whilst I'm in the snipping tool mode. I can show a screenshot if it helps?

**Comment by Lailloken** [comment]
> Do you know if there's any way to make it run when launching PoE with Steam?

I have seen Reddit comments where people said they created batch-files to launch PoE and 3rd-party tools. I thought about adding an option along the lines of "launch the game-client on script-start" but I have not thought it through yet.

> The UI reports "Screen cap failed" whilst I'm in the snipping tool mode. I can show a screenshot if it helps?
> 
> Edit: Screenshot: https://imgur.com/6iYHNyv - Using the Stash tab to show but the same was happening for the skilltree. I think I was just quick enough once? Not 100% sure.

Which Windows version are you running, and is it maybe a custom install? Someone reported a similar issue once and they were running an LTSC version of Windows which uses a different snipping tool.

When you click "cal", the script waits for the snipping tool to be active -- and once that closes, it knows snipping was either completed or aborted. It times out if it doesn't find the snipping tool after two seconds.

To check if your Windows is using a different snipping tool, could you right-click the script's tray-icon and open "window spy"?
- Then check the box in the top right that says "follow mouse"
- The Window spy will now show the stats of the window underneath the cursor, and pause while holding Shift or Control
- open the snipping tool, hold shift or control, and close the snipping tool without letting go of the key
- the window spy should now show the stats of the snipping tool's window, I need a screenshot of that
  ![image](https://github.com/Lailloken/Lailloken-UI/assets/61888437/08c9886a-71d9-480c-956c-2557c523ec60)

**Comment by zefrogi** [comment]
> I have seen Reddit comments where people said they created batch-files to launch PoE and 3rd-party tools. I thought about adding an option along the lines of "launch the game-client on script-start" but I have not thought it through yet.

Yeah, I tried to make and got PoE to open but not the LaiUI. Not biggie though! Still love it.

> Which Windows version are you running, and is it maybe a custom install? Someone reported a similar issue once and they were running an LTSC version of Windows which uses a different snipping tool.

I don't _think_ I'm using a wild version of Windows. I have managed to do the snipping before and did get the skilltree capture to work. Screenshot: https://imgur.com/a/75zUiwE

**Comment by Lailloken** [comment]
> I don't _think_ I'm using a wild version of Windows. I have managed to do the snipping before and did get the skilltree capture to work. Screenshot: https://imgur.com/a/75zUiwE

Hmm yeah, it's a different snipping tool. Maybe a Windows update changed the default snipping tool at some point. I think I'll just try to find a way to make the snipping detection exe-agnostic because it's not unlikely it'll change again in Windows.

For the time being, you can open the main Lailloken UI.ahk file with an editor and search for
```
GroupAdd, snipping_tools, ahk_exe ScreenClippingHost.exe
GroupAdd, snipping_tools, ahk_exe ShellExperienceHost.exe
```

and then change one of the EXEs to the one from your screenshot.
***
<br>

As for your main problem of the leveling overlay not showing up in v1.52.1:
- Are you using a multi-monitor setup? It's not likely, but maybe the overlay is showing somewhere outside your main screen.

- you could try manually editing the `ini\leveling tracker.ini` file and set custom coordinates, maybe the default ones are somehow broken (even if I can't replicate that)
  - under the `[settings]` section edit/add the lines `x-coordinate=500` and `y-coordinate=500`, then save it and restart the script
  - this should make it appear near the top-left corner of your main screen

**Comment by zefrogi** [comment]
> and then change one of the EXEs to the one from your screenshot.

Champion! Worked as expected after the change of the exe.

> Are you using a multi-monitor setup? It's not likely, but maybe the overlay is showing somewhere outside your main screen.

Indeed! That is the issue. I swapped PoE to main screen and the levelling UI appeared. Oddly, it didn't like when I tried to put PoE back on my second monitor. It stayed on the first.

Now, however, when I open PoE on the second. The levelling UI now appears with the game on the second screen. So does it need to sync with PoE on the main and then get clamped?

Working now regardless so thanks for your help!

**Comment by Lailloken** [comment]
Reopened for visibility: I can confirm there's an oversight in the guide-overlay's positioning on non-primary monitors.

Temporary fix:
- manual changes to the `ini\leveling tracker.ini` file mentioned above, then saving the file and restarting the script

-  this will make the overlay appear on the correct screen at least, and the overlay can be moved by dragging the small white square
![img](https://private-user-images.githubusercontent.com/61888437/302258169-1912c91b-31ca-4427-8419-60b776e67703.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDgxNzk5OTYsIm5iZiI6MTcwODE3OTY5NiwicGF0aCI6Ii82MTg4ODQzNy8zMDIyNTgxNjktMTkxMmM5MWItMzFjYS00NDI3LTg0MTktNjBiNzc2ZTY3NzAzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjE3VDE0MjEzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM5YzE0NTI0ZGZmNGY1MGEzZmVhY2ZiODAzOTMwNDdlMWFlOTkyYWQ4YmE2NThmMzE4MWFiYzk5NTJkYTk2ODcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.I82JBTwQ_zTHRuxaPY6d4k34bvtDOmG4eCiSszXnuPE)

**Comment by Lailloken** [comment]
Both issues should be fixed in v1.52.2
